community/kernel/src/main/java/org/neo4j/kernel/internal/GraphDatabaseAPI.java

```diff
@@ -48,6 +52,43 @@
      */
     URL validateURLAccess( URL url ) throws URLAccessValidationError;
 
-    @Deprecated
     String getStoreDir();
+
+    /**
+     * Returns all relationship types currently in the underlying store.
+     * Relationship types are added to the underlying store the first time they
+     * are used in a successfully commited {@link Node#createRelationshipTo
+     * node.createRelationshipTo(...)}. Note that this method is guaranteed to
+     * return all known relationship types, but it does not guarantee that it
+     * won't return <i>more</i> than that (e.g. it can return "historic"
+     * relationship types that no longer have any relationships in the node
+     * space).
+     *
+     * @return all relationship types in the underlying store
+     */
+    ResourceIterable<RelationshipType> getAllRelationshipTypesInUseAndNot();
```

Comment:
@tinwelint now I'm super confused. it does seem to me that GDS#getAllXX methods are returning (2) instead of (1). this is the call chain i follow:
https://github.com/neo4j/neo4j/blob/3.1/community/kernel/src/main/java/org/neo4j/kernel/impl/factory/GraphDatabaseFacade.java#L397
https://github.com/neo4j/neo4j/blob/3.1/community/kernel/src/main/java/org/neo4j/kernel/impl/factory/GraphDatabaseFacade.java#L415
https://github.com/neo4j/neo4j/blob/3.1/community/kernel/src/main/java/org/neo4j/kernel/impl/api/TokenAccess.java#L109
https://github.com/neo4j/neo4j/blob/3.1/community/kernel/src/main/java/org/neo4j/kernel/impl/api/TokenAccess.java#L149
https://github.com/neo4j/neo4j/blob/3.1/community/kernel/src/main/java/org/neo4j/kernel/impl/api/TokenAccess.java#L159
https://github.com/neo4j/neo4j/blob/3.1/community/kernel/src/main/java/org/neo4j/kernel/impl/api/TokenAccess.java#L57

am i missing something?

`getAllRelationshipTypesInUseAndNot` method is used in `DatabaseMetadataService` where users can actually ask for tokens not used for anything in user's sense.